### PR TITLE
Local Dashboard fixes

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Fix most of https://github.com/enso-org/cloud-v2/issues/1459
  - Prevent click + click from triggering rename on Windows and Linux. Behavior is preserved on macOS.
  - Fix text in Drive when root folder is empty
  - Properly remove the "Drop here to upload box" after a file is dropped
  - "Copy as path" now unconditionally uses `/` for path delimiters, even on Windows
  - Duplicating a project in the root folder on Windows no longer errors
  - Extra folders in the sidebar now (correctly) show folder name, rather than path, on Windows
  - Mouse pointer when dragging to a folder is now move, not copy

Not addressed:
- [no-repro] Tooltips should have some latency before showing up
  - This should already be the case, although it may work weirdly (once the tooltip opens, there is no delay on subsequent tooltips opening until the last tooltip closes.)
- [no-repro] Ctrl-click should add to selection
- Column width should be resizable
  - This requires a refactor and therefore is considered out of scope for this PR
- [no-repro] Choosing root folder needs a file browser
- [no-repro] Choosing root folder doesn't do anything get the same list as we had before
- [no-repro] Open in explorer didn't work in a file but did on project - possibly fixed by path changes.
- [no-repro] Opening an enso-project by double clicking resulted on it being renamed with a (2)

Related changes:
- Make "root directory" picker's file browser default to the current root directory

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

**Comment by PabloBuchu** [comment]
| [no-repro] Ctrl-click should add to selection
Does it mean it works for you? I also can't select multiple things with cmd/ctrl maybe key mapping is bugged?

| Opening an enso-project by double clicking resulted on it being renamed with a (2)
Looks bugged to me but maybe this needs a separate issue?

https://github.com/user-attachments/assets/d0a0ddf2-7d4f-4a6e-bbff-0c9ccd13327d

**Comment by somebody1234** [comment]
> I also can't select multiple things with cmd/ctrl maybe key mapping is bugged?

maybe - editing an asset's name uses the same shortcut (cmd+click) - does that work for you?

**Comment by somebody1234** [comment]
ahhh, double clicking like that...

**Review by PabloBuchu — APPROVED** [review]

**Comment by PabloBuchu** [comment]
✅

**Comment by somebody1234** [comment]
@PabloBuchu some minor re-QA might be needed, for the double clicking to open. steps to repro:
- set your Local root directory to a *non*-default value
- double click a file. currently errors on develop since the file was uploaded to the *default* root directory so it is not present in the FE's root directory.
- also refresh to make sure we don't repeatedly upload the new file every time we refresh (due to changes to when the file is uploaded.)

**Comment by PabloBuchu** [comment]
@somebody1234 I think looks good

File: app/dashboard/src/pages/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -145,6 +147,30 @@ function DashboardInner(props: DashboardProps) {
   const openProjectMutation = projectHooks.useOpenProjectMutation()
   const renameProjectMutation = projectHooks.useRenameProjectMutation()
 
+  useQuery({
```
Comment (MrFlashAccount) [inline app/dashboard/src/pages/dashboard/Dashboard.tsx]: I think it's better to use `usePrefetchQuery` here: https://stackoverflow.com/a/77486960

**Review by MrFlashAccount — APPROVED** [review]
LGTM

File: app/ide-desktop/common/package.json [hunk 2]
```diff
@@ -28,14 +28,14 @@
     "test": "vitest run"
   },
   "peerDependencies": {
-    "@tanstack/query-core": "5.45.0",
-    "@tanstack/vue-query": ">= 5.45.0 < 5.46.0"
+    "@tanstack/query-core": "5.54.1",
+    "@tanstack/vue-query": ">= 5.54.0 < 5.56.0"
```
Comment (kazcw) [inline app/ide-desktop/common/package.json:32]: With this update, `vueQuery.ts` in `enso-common` is no longer needed--the fix it implements has been merged upstream, so we can use the standard `vueQuery.QueryClient`.

**Review by kazcw — APPROVED** [review]
Based on the tanstack-query changelogs, this version update shouldn't cause any problems for the GUI

**Pull request merged**